libraries/AP_InertialSensor/AP_InertialSensor_SCHA63T.cpp

```diff
@@ -0,0 +1,456 @@
+    hal.scheduler->delay(405);
+
+    // start EOI = 1
+    ret_scha63t &= write_register(SCHA63T_UNO, RESCTRL, RES_EOI);
+    ret_scha63t &= write_register(SCHA63T_DUE, RESCTRL, RES_EOI);
+
+    // first read summary status
+    ret_scha63t &= read_register(SCHA63T_UNO, S_SUM, val);
+    ret_scha63t &= read_register(SCHA63T_DUE, S_SUM, val);
+    // 2.5ms or more
+    hal.scheduler->delay(3);
+
+    // second read summary status
+    ret_scha63t &= read_register(SCHA63T_UNO, S_SUM, val);
+    ret_scha63t &= read_register(SCHA63T_DUE, S_SUM, val);
+    // 2.5ms or more
+    hal.scheduler->delay(3);
+
+    // read summary status
+    ret_scha63t &= read_register(SCHA63T_UNO, S_SUM, val);
+    // check UNO summary status
+    if (!((val[1] & 0x9e) && (val[2] & 0xda))) {
+        read_summary_error = true;
+    }
+    ret_scha63t &= read_register(SCHA63T_DUE, S_SUM, val);
+    // check DUE summary status
+    if (!((val[1] & 0xf8) && (val[2] & 0x03))) {
+        read_summary_error = true;
+    }
+
+    // check error
+    if (read_summary_error) {
+        return false;
+    }
+
+    return true;
+}
+
+/*
+  read accel fifo
+ */
+void AP_InertialSensor_SCHA63T::read_accel(void)
+{
+    uint8_t rsp_accl_x[4];
+    uint8_t rsp_accl_y[4];
+    uint8_t rsp_accl_z[4];
+    uint8_t rsp_temper[4];
+
+    int16_t accel_x = 0;
+    int16_t accel_y = 0;
+    int16_t accel_z = 0;
+    int16_t uno_temp = 0;
+
+    // ACCL_X Cmd Send (first response is undefined data)
+    ret_scha63t &= read_register(SCHA63T_UNO, ACC_X, rsp_accl_x);
+    // ACCL_Y Cmd Send + ACCL_X Response Receive
+    ret_scha63t &= read_register(SCHA63T_UNO, ACC_Y, rsp_accl_x);
+    // ACCL_Z Cmd Send + ACCL_Y Response Receive
+    ret_scha63t &= read_register(SCHA63T_UNO, ACC_Z, rsp_accl_y);
+    // TEMPER Cmd Send + RATE_X Response Receive
+    ret_scha63t &= read_register(SCHA63T_UNO, TEMP, rsp_accl_z);
+    // TEMPER Cmd Send + TEMPRE Response Receive
+    ret_scha63t &= read_register(SCHA63T_UNO, TEMP, rsp_temper);
+
+    // response data address check
+    if (((rsp_accl_x[0] & 0x7C) >> 2) == ACC_X) {
+        accel_x = combine(rsp_accl_x[1], rsp_accl_x[2]);
+    } else {
+        ret_scha63t &= false;
+    }
+    if (((rsp_accl_y[0] & 0x7C) >> 2) == ACC_Y) {
+        accel_y = combine(rsp_accl_y[1], rsp_accl_y[2]);
+    } else {
+        ret_scha63t &= false;
+    }
+    if (((rsp_accl_z[0] & 0x7C) >> 2) == ACC_Z) {
+        accel_z = combine(rsp_accl_z[1], rsp_accl_z[2]);
+    } else {
+        ret_scha63t &= false;
+    }
+    if (((rsp_temper[0] & 0x7C) >> 2) == TEMP) {
+        uno_temp = combine(rsp_temper[1], rsp_temper[2]);
+    } else {
+        ret_scha63t &= false;
+    }
+    set_temperature(accel_instance, uno_temp);
+
+    // change coordinate system from left hand too right hand
+    accel_z = (accel_z == INT16_MIN) ? INT16_MAX : -accel_z;
+
+    Vector3f accel(accel_x, accel_y, accel_z);
+    accel *= (GRAVITY_MSS / 4905.f); // 4905 LSB/g, 0.204mg/LSB
```

Comment:
0.204 is 4902 not 4905

Reply:
@jfbblue0922 any chance you could double check whether this should be 4905 or 4902?

Reply:
According to the datasheet, 4905 is correct.
We should change the comment 0.204 mg/LSB to 0.20387 mg/LSB.
![fig](https://github.com/ArduPilot/ardupilot/assets/113963003/4106dfd8-52b5-4aec-94ee-61360e5a9e7d)

Reply:
comment updated, thanks